### PR TITLE
Option page fixes

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Some fixes around formatting options.

* Make sure the settings correctly roam and are shown
* Make sure radio buttons actually work how you think they do

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
Some reviewers: @Pilchie, @rchande

**Comment by rchande** [comment]
Does recreating the ViewModel mean that the each section's (Indentation, etc) preview gets reset to the first item every time your settings sync?

Otherwise LGTM.

**Comment by jasonmalinowski** [comment]
The ViewModel will only be recreated the first time the page is gone to in an Tools > Options session. Since there's no real selected item at that point, it seemed safe.

**Comment by rchande** [comment]
Does LoadSettings only get called once?

**Comment by jasonmalinowski** [comment]
It's called once per page per Tools > Options session. So opening it, change a few options, hitting cancel, and opening it again will trigger another LoadSettings.

(Come to think of it, was cancelling and reopening the dialog broken too?)

**Comment by rchande** [comment]
That sounds fine. Is it possible/do we care about settings getting synced while the page is open?

**Comment by jasonmalinowski** [comment]
I guess the only scenario that might matter is if it syncs while open, does pressing OK blow away the things that synced during that time? It used to be the underlying SetOptions call tried to avoid side-effects and only affect the options directly changed, but it's been a long time since I wrote that.

**Comment by DustinCampbell** [comment]
:+1:

**Comment by jasonmalinowski** [comment]
Information for shiproom:

*Customer Impact:* this pull request fixes several distinct issues with our formatting options:

1. Radio button-ed options, of which there are two, do not work correctly -- choosing an option applies not the one you picked but the one you previously picked. Imagine there is an formatting option with three possible choices, A, B, and C. Without the fix, if you had A selected and choose B, the preview for *A* is shown and *A* is actually the option that will get applied if you hit OK. If you then selected A again, the preview for *B* is shown and *B* will be applied. The user workaround for this bug would be to choose the option they want, and then choose something else.
2. Pressing cancel in Tools > Options doesn't forget your choices. If you went to Tools > Options, changed a few C# formatting settings and change your mind by pressing cancel, the next time you open the dialog your abandoned settings will still be shown, as opposed to the settings that you're actually using. The only workaround to viewing your effective options would be to restart Visual Studio.
3. Most C# formatting settings will not roam until you restart Visual Studio. We are internally throwing exceptions on a extensibility point so there might be unknown side effects of that.
4. If settings are changed via some external mechanism (either setting sync, automation, Tools > Reset Settings.) the new formatting options aren't shown in Tools > Options. Once again, the workaround would be to restart Visual Studio.

For all bugs but the first, restarting VS would be sufficient to work around the bug. For the first, the workaround is to select the option you want and then *select something else*, something the customer really won't figure out unless we publish it.

Issue 1, 2, and 4 are regressions from VS2013. 3 is new since this is new code to support the new setting sync infrastructure being introduced in VS2015. 4 is a bug you can hit without setting sync, but setting sync would probably be the primary contributor to it.

*Risk:* no change is particularly high risk. Primary concern would simply be this is a relatively forgotten part of our codebase, as evidenced by the bugs.

**Comment by pharring** [comment]
Approved 6/5 ML Shiproom pending buddy testing.

**Comment by jasonmalinowski** [comment]
@rchande was nominated for buddy testing. Ravi, how's it coming?

**Comment by rchande** [comment]
@jasonmalinowski 
Radio buttons actually work the way someone might expect now :smile:.
Canceling and restarting the options page shows your current settings, not whatever the page last showed.
Roaming works.
If I reset settings in one instance, the option page reflects the change in the other.

LGTM

File: src/VisualStudio/Core/Impl/Options/OptionPreviewControl.xaml.cs [hunk 1]
```diff
@@ -73,6 +71,13 @@ internal override void SaveSettings()
 
         internal override void LoadSettings()
         {
+            this.ViewModel = _createViewModel(this.OptionService.GetOptions(), _serviceProvider);
+
+            // Use the first item's preview.
+            var firstItem = this.ViewModel.Items.OfType<CheckBoxOptionViewModel>().First();
```
Comment (Pilchie) [inline src/VisualStudio/Core/Impl/Options/OptionPreviewControl.xaml.cs]: Does this mean well crash if we ever have a page with no check boxes?

**Comment by Pilchie** [comment]
:+1:

File: src/VisualStudio/Core/Impl/Options/OptionPreviewControl.xaml.cs (hunk 1 above)
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Impl/Options/OptionPreviewControl.xaml.cs]: Yes, so let's not do that.

**Pull request merged**